src/test/passes/arithmetic/xor-x86_64-darwin.c

```diff
@@ -0,0 +1,98 @@
+// REQUIRES: x86-registered-target
+
+// RUN:                                        clang -target x86_64-apple-darwin                         -O1 -S %s -o - | FileCheck --check-prefix=R0 %s
+// RUN: env OMVLL_CONFIG=%S/config_rounds_0.py clang -target x86_64-apple-darwin -fpass-plugin=%libOMVLL -O1 -S %s -o - | FileCheck --check-prefix=R0 %s
+// RUN: env OMVLL_CONFIG=%S/config_rounds_1.py clang -target x86_64-apple-darwin -fpass-plugin=%libOMVLL -O1 -S %s -o - | FileCheck --check-prefix=R1 %s
+// RUN: env OMVLL_CONFIG=%S/config_rounds_2.py clang -target x86_64-apple-darwin -fpass-plugin=%libOMVLL -O1 -S %s -o - | FileCheck --check-prefix=R2 %s
+// RUN: env OMVLL_CONFIG=%S/config_rounds_3.py clang -target x86_64-apple-darwin -fpass-plugin=%libOMVLL -O1 -S %s -o - | FileCheck --check-prefix=R3 %s
+
+// R0-LABEL: _memcpy_xor:
+// R0:       LBB0_2:
+// R0:           movzbl	(%rsi,%rcx), %edx
+// R0:           xorb	$35, %dl
+// R0:           movb	%dl, (%rdi,%rcx)
+// R0:           addq	$1, %rcx
+// R0:           cmpq	%rcx, %rax
+// R0:           jne	LBB0_2
+
+// FIXME: It seems like rounds=1 doesn't actually do anything!
```

Comment:
Is that expected behavior?

Reply:
Not really but the current MBA expressions can be simplified by LLVM [^1]. There is also a statement about this limitation in the documentation: https://obfuscator.re/omvll/passes/arithmetic/.

[^1]: [InstCombineAndOrXor.cpp](https://github.com/llvm/llvm-project/blob/fe01c084249969518bcd69aa2fedffc9f017f43a/llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp#L1703-L1708)